### ISSUE TITLE
First call to createDiagnosticsAdapter and YamlDocuments.ensureCache throws null reference error

### ISSUE THREAD

**Issue by hillin** [remcohaszing/monaco-yaml#111]
A quick debugging shows that only the first call to these functions throws error; they are both tied to a null *inmemory* model with uri `inmemory://model/1`. However at this point the document is already loaded.

```
> monaco.contribution.js:58 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'getModeId')
    at doValidate (monaco.contribution.js:58)
doValidate @ monaco.contribution.js:58
async function (async)
doValidate @ monaco.contribution.js:54
onModelAdd @ monaco.contribution.js:72
fire @ event.js:424
createModel @ modelServiceImpl.js:324
doCreateModel @ standaloneCodeEditor.js:323
createTextModel @ standaloneCodeEditor.js:317
StandaloneEditor @ standaloneCodeEditor.js:180
(anonymous) @ standaloneEditor.js:62
withAllStandaloneServices @ standaloneEditor.js:49
create @ standaloneEditor.js:61
...

> TypeError: Cannot read properties of null (reading 'uri')
    at YamlDocuments.ensureCache (yaml.worker.worker.js:23425)
    at YamlDocuments.getYamlDocument (yaml.worker.worker.js:23418)
    at Object.findLinks2 [as findLinks] (yaml.worker.worker.js:24862)
    at Object.findLinks (yaml.worker.worker.js:25404)
    at EditorSimpleWorker.fmr (yaml.worker.worker.js:14912)
    at SimpleWorkerServer._handleMessage (yaml.worker.worker.js:12526)
    at Object.handleMessage (yaml.worker.worker.js:12512)
    at SimpleWorkerProtocol._handleMessage (yaml.worker.worker.js:12381)
    at SimpleWorkerProtocol.handleMessage (yaml.worker.worker.js:12354)
    at SimpleWorkerServer.onmessage (yaml.worker.worker.js:12516)
    at errors.js:8
```

**Comment by hillin** [comment]
Probably related to https://github.com/remcohaszing/monaco-yaml/issues/106

**Comment by hillin** [comment]
OK this has something to do with the way I create the editor, I create the editor and call `setDiagnosticsOptions` first, then after a while (after the source is loaded), I create the model.
The issue disappears if I create the model before calling `setDiagnosticsOptions`.

Is this an expected behavior? I'll close this issue if so.

**Comment by remcohaszing** [comment]
What version of `monaco-yaml` are you using?

**Comment by hillin** [comment]
> What version of `monaco-yaml` are you using?

3.2.0

**Comment by remcohaszing** [comment]
> > What version of `monaco-yaml` are you using?
> 
> 3.2.0

Then I think you ran into the issue you linked, which was fixed in 3.2.1.

**Comment by hillin** [comment]
Great, thanks!
